src/uri/uri.cc

```diff
@@ -249,6 +249,34 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto parsed_path = URI{std::string{path}};
+  const std::string path_to_set = parsed_path.path().value();
```

Comment:
This feels like an unnecessary extra copy.

Reply:
Actually same comment as below. I think it can all be simplified assuming you are already parsing the path in the line above? You can then just copy the `path` member, right?